src/shared/components/Gallery/Gallery.tsx

```diff
@@ -3,30 +3,46 @@ import React, { ComponentProps, useRef } from 'react'
 import { Arrow } from '@/shared/components/Carousel/Carousel.style'
 import { SvgGlyphChevronLeft, SvgGlyphChevronRight } from '@/shared/icons'
 
-import { CarouselArrowsContainer, Container, HeadingContainer } from './Gallery.style'
+import {
+  CarouselArrowsContainer,
+  Container,
+  HeadingContainer,
+  StyledSvgGlyphPlay,
+  TitleWrapper,
+  WatchAllLink,
+} from './Gallery.style'
 
 import { Carousel, CarouselProps } from '../Carousel/Carousel'
 import { Text } from '../Text'
 
 export type GalleryProps = {
   title?: string
   className?: string
+  watchAllUrl?: string
 } & CarouselProps
 
 type ImperativeHandleData = {
   getPrevArrowProps: () => ComponentProps<typeof Arrow>
   getNextArrowProps: () => ComponentProps<typeof Arrow>
 }
 
-export const Gallery: React.FC<GalleryProps> = ({ title, className, ...carouselProps }) => {
+export const Gallery: React.FC<GalleryProps> = ({ title, className, watchAllUrl, ...carouselProps }) => {
   // TODO: this is the only place in the app that requires refs to buttons. Once we refactor this component, we can remove forwardRef from buttons
   const prevArrowRef = useRef<HTMLButtonElement>(null)
   const nextArrowRef = useRef<HTMLButtonElement>(null)
   const carouselRef = useRef<ImperativeHandleData>(null)
   return (
     <Container className={className}>
       <HeadingContainer>
-        {title && <Text variant="h4">{title}</Text>}
+        <TitleWrapper>
+          {title && <Text variant="h4">{title}</Text>}
+          {watchAllUrl && (
+            <WatchAllLink to={watchAllUrl}>
+              <StyledSvgGlyphPlay width={14} height={14} />
+              Watch all
```

Comment:
Maybe we should change it to `See all` for now. `Watch all` suggests that it will run some kind of playlist and we don't have that functionality now.

@kdembler  wdyt?

Reply:
Agreed!